src/java/htsjdk/samtools/filter/InsertSizeFilter.java

```diff
@@ -0,0 +1,31 @@
+package htsjdk.samtools.filter;
+
+import htsjdk.samtools.SAMException;
+import htsjdk.samtools.SAMRecord;
+
+/**
+ * Filter things that fall outside a specified range of insert sizes.
+ * This will automatically omit unpaired reads.
+ */
+public class InsertSizeFilter implements SamRecordFilter {
+    final int minInsertSize;
+    final int maxInsertSize;
+
+    public InsertSizeFilter(final int minInsertSize, final int maxInsertSize) {
+        if (minInsertSize > maxInsertSize) throw new SAMException("Cannot have minInsertSize > maxInsertSize");
```

Comment:
That's in the CLP, which allows you to set max=0 for no upper bound, which then passes Integer.MAX_VALUE into the InsertSizeFilter.

I felt that the "max=0" behavior was too specific/subtle for HTSJDK - if you're calling the filter directly, give it Integer.MAX_VALUE if you want to upper bound.

Does that sound reasonable?